lib/install/management/create.go

```diff
@@ -120,100 +121,85 @@ func (d *Dispatcher) uploadImages(files map[string]string) error {
+			// check iso first
+			d.op.Debugf("Checking if file already exists: %s", isoTargetPath)
+			_, err := ds.Stat(d.op, isoTargetPath)
+			if err != nil {
+				switch err.(type) {
+				case object.DatastoreNoSuchFileError:
+					d.op.Debug("File not found. Nothing to do.")
+				case object.DatastoreNoSuchDirectoryError:
+					d.op.Debug("Directory not found. Nothing to do.")
+				default:
+					d.op.Debugf("ISO file already exists, deleting: %s", isoTargetPath)
+					err := fm.Delete(d.op, isoTargetPath)
+					if err != nil {
+						d.op.Debugf("Failed to delete image (%s) with error (%s)", image, err.Error())
+						return err
 					}
 				}
-
-				return d.session.Datastore.UploadFile(d.op, image, path.Join(d.vmPathName, key), nil)
 			}
 
-			// counter for retry decider
-			retryCount := uploadRetryLimit
+			d.op.Infof("Uploading %s as %s", baseName, key)
 
-			// decider for our retry, will retry the upload uploadRetryLimit times
-			uploadRetryDecider := func(err error) bool {
-				if err == nil {
-					return false
-				}
+			ul := progresslog.NewUploadLogger(d.op.Infof, baseName, time.Second*3)
+			// need to wait since UploadLogger is asynchronous.
+			defer ul.Wait()
 
-				retryCount--
-				if retryCount < 0 {
-					d.op.Warnf("Attempted upload a total of %d times without success, Upload process failed.", uploadRetryLimit)
-					return false
-				}
-				d.op.Warnf("failed an attempt to upload isos with err (%s), %d retries remain", err.Error(), retryCount)
-				return true
-			}
+			return d.session.Datastore.UploadFile(d.op, image, path.Join(d.vmPathName, key),
+				progresslog.UploadParams(ul))
+		}
 
-			// Build retry config
-			backoffConf := retry.NewBackoffConfig()
-			backoffConf.InitialInterval = uploadInitialInterval
-			backoffConf.MaxInterval = uploadMaxInterval
-			backoffConf.MaxElapsedTime = uploadMaxElapsedTime
-
-			uploadErr := retry.DoWithConfig(operationForRetry, uploadRetryDecider, backoffConf)
-			if uploadErr != nil {
-				finalMessage = fmt.Sprintf("\t\tUpload failed for %q: %s\n", image, uploadErr)
-				if d.force {
-					finalMessage = fmt.Sprintf("%s\t\tContinuing despite failures (due to --force option)\n", finalMessage)
-					finalMessage = fmt.Sprintf("%s\t\tNote: The VCH will not function without %q...", finalMessage, image)
-					results <- errors.New(finalMessage)
-				} else {
-					results <- errors.New(finalMessage)
-				}
+		// counter for retry decider
+		retryCount := uploadRetryLimit
+
+		// decider for our retry, will retry the upload uploadRetryLimit times
+		uploadRetryDecider := func(err error) bool {
+			if err == nil {
+				return false
 			}
-			wg.Done()
-		}(key, image)
-	}
 
-	wg.Wait()
-	close(results)
+			retryCount--
+			if retryCount < 0 {
+				d.op.Warnf("Attempted upload a total of %d times without success, Upload process failed.", uploadRetryLimit)
+				return false
+			}
+			d.op.Warnf("failed an attempt to upload isos with err (%s), %d retries remain", err.Error(), retryCount)
```

Comment:
While this string has been present before this change, it would be a good idea to refine it since we're making changes here anyway. Suggest replacing `failed` with `Failed` to maintain consistent casing for log messages (other log messages start with uppercase).